public/app/features/explore/state/selectors.ts

```diff
@@ -23,3 +25,43 @@ export const isLeftPaneSelector = (exploreId: string) =>
 export const getExploreItemSelector = (exploreId: string) => createSelector(selectPanes, (panes) => panes[exploreId]);
 
 export const selectCorrelationDetails = createSelector(selectExploreRoot, (state) => state.correlationEditorDetails);
+
+export const selectShowQueryHistory = createSelector(selectExploreRoot, (state) => state.showQueryHistory);
+
+export const selectExploreDSMaps = createSelector(selectPanesEntries, (panes) => {
+  const exploreDSMap = panes
+    .map((pane) => {
```

Comment:
You can also expand the param to make the code more readable `.map(([exploreId, pane]) => {...})`